arangodb-foxx-services/src/main/model.ts

```diff
@@ -15,4 +15,57 @@
  */
 
 export type DocumentKey = string
-export type ExecutionEvent = object
+export type ExecutionEvent = ExecutionEventLike
+export type DataSource = DataSourceLike
+
+interface DataSourceLike {
+    _key: string
+}
+
+interface ExecutionEventExtra {
+    appId: string
+}
+
+export interface ExecutionEventLike {
+    timestamp: bigint
+    extra: ExecutionEventExtra
+}
+
+/**
+ * mimics backend's [[za.co.absa.spline.consumer.service.model.LineageOverview]]
+ */
+export class LineageOverview {
+    readonly info: LineageOverviewInfo
+    readonly graph: LineageOverviewGraph
+
+    constructor(info: LineageOverviewInfo, graph: LineageOverviewGraph) {
+        this.info = info
+        this.graph = graph
+    }
+}
+
+export class LineageOverviewInfo {
+    readonly timestamp: bigint
+    readonly applicationId: string
+    readonly targetDataSourceId: string
+
+    constructor(timestamp: bigint, applicationId: string, targetDataSourceId: string) {
+        this.timestamp = timestamp
+        this.applicationId = applicationId
+        this.targetDataSourceId = targetDataSourceId
+    }
+}
+
+export class LineageOverviewGraph {
+    readonly depthRequested: number
+    readonly depthComputed: number
+    readonly nodes: Array<object>
+    readonly edges: Array<object>
```

Comment:
Can we use algebraic data types instead of those objects?

Reply:
Thanks for the tip and explanation, done.